---
title: "NYC Data analysis"
author: "Tim Read"
date: "August 31, 2015"
output: html_document
---

Analysis of NYC subway data.  Based on a older script called 'NYsubwaySa.R'.

```{r setup}
print(date())
library(RgoogleMaps)
library(dplyr)
library(vegan)
library(ade4)
library(e1071)
library(gtools)
#library(biomod2)
library(RColorBrewer)
library(assertthat)
source('./staph_metagenome_tools.R')

```

### Load files

```{r, }
#Public data from the original NYC subway publication: Afshinnekoo E, Meydan C, Chowdhury S, Jaroudi D, Boyer C, Bernstein N, Maritz JM, Reeves D, Gandara J, Chhangawala S, Ahsanuddin S, Simmons A, Nessel T, Sundaresh B, Pereira E, Jorgensen E, Kolokotronis S-O, Kirchberger N, Garcia I, Gandara D, Dhanraj S, Nawrin T, Saletore Y, Alexander N, Vijay P, Hénaff EM, Zumbo P, Walsh M, O’Mullan GD, Tighe S, Dudley JT, Dunaif A, Ennis S, O’Halloran E, Magalhaes TR, Boone B, Jones AL, Muth TR, Paolantonio KS, Alter E, Schadt EE, Garbarino J, Prill RJ, Carlton JM, Levy S, Mason CE. Geospatial Resolution of Human and Bacterial Diversity with City-Scale Metagenomics. Cell Systems [Internet]. Elsevier; 2015 Jul 29;1(1):72–87.
NYCdata <- (read.csv("./Data/DataTable5-metaphlan-metadata_v19 .csv",stringsAsFactors = FALSE, header = TRUE))[1:4]
# we made this table from parsing the SRA
strain_SRA <- read.table("./Data/runs-to-samples.txt", header = TRUE)
colnames(strain_SRA) <- c("Run", "Sample.ID") #to make join easier
NYCdata_SRA <- left_join(NYCdata, strain_SRA, by = "Sample.ID")


```

```{r}
#Our table of binstrain results
# Staph_betas <- read.csv("./Data/Final_Beta_NYC_STAPH.csv",stringsAsFactors = FALSE, header = TRUE)
# #remove the final row, which is actually a bead wash control (suggests some carrover contamination)
# Staph_betas <- Staph_betas[-130,]



```

### import CC, merge CC, rename and tidy, create various forms of the data to be fed into funciotns 
(reflects the piecemeal development of the fucnitons over several months with changing specs)

(Note used some similar commands in HMP_coverage.Rmd).  Need to merge becasue we originally separated CC30, CC8 and CC5 into multiple groups but this turned out not to be specific.

```{r maps}
Staph_betas <- read.csv("./Data/FInal_NYC_Staph_MecA_Table.csv", stringsAsFactors = FALSE, header = TRUE) %>% filter(Staph_Coverage > 0.025) %>% filter(!(Sample.Id == "SRR1750088")) 
colnames(Staph_betas)[colnames(Staph_betas) == "CC_8_72"] <- "CC_72"
Staph_betas <- merge_CCs(Staph_betas,"CC_8_")
Staph_betas <- merge_CCs(Staph_betas,"CC_30_")
Staph_betas <- merge_CCs(Staph_betas,"CC_5_")

names(Staph_betas)[names(Staph_betas) == "MLST_93"] <- "CC_93"
Staph_betas <- Staph_betas[,mixedsort(colnames(Staph_betas))]
colnames(Staph_betas) <- gsub("(CC_.{0,3})_.{0,4}$","\\1",colnames(Staph_betas))
colnames(Staph_betas)[colnames(Staph_betas) == "Sample.Id"] <- "Run"
#remove SRR1748847, which has an incorrect coord
Staph_betas <- filter(Staph_betas, Run != "SRR1748847")
staph_mat <- make_subtype_matrix(Staph_betas)

staph_df <- as.data.frame(staph_mat)
staph_df <- cbind(staph_df,Staph_betas$Run)
colnames(staph_df)[colnames(staph_df) == "Staph_betas$Run"] <- "Run"
staph_df$Run <- as.factor(staph_df$Run)
staph_df_coords <- inner_join(NYCdata_SRA, staph_df, by = "Run") #one of the samples does not have a lat , lon posiiotn

coords_df_subtype_mat <- make_subtype_matrix(staph_df_coords)
```

### genotype plots

```{r NYC_genotypes}

presence_mat <- as.data.frame(bintr(staph_mat,0.2))
top_score_mat <- as.data.frame(bintr(staph_mat,0.5))
# png("~/Dropbox/ARTICLES_BY_TDR/2015-staph-metagenome/HMP_barchart.png",width=640, height =640, res = 75)
# dev.off()
all_genotypes_plot(presence_mat,"All NYC samples, subtypes present > 0.2")
all_genotypes_plot(top_score_mat,"All NYC samples, subtypes present > 0.5")
```

### Color stations reporting S. aureus
```{r}
locations_with_runs <- filter(NYCdata_SRA, !is.na(Run) ) %>%
  select(Latitude, Logitude)

lats <- as.numeric(as.character(locations_with_runs$Latitude))
lons <- as.numeric(as.character(locations_with_runs$Logitude))
cols_staph <- rep(NULL,length(lats))
#all the statiitons which had S. aureus
cols_staph[which(NYCdata_SRA$Run %in% Staph_betas$Run)] <- "red"

alllats <- as.numeric(as.character(NYCdata$Latitude)) #every non-numeric value is converted to NA
alllons <- as.numeric(as.character(NYCdata$Logitude))
```
### Generate google maps
```{r}
gmap1 <- GetMap(center = c(lat = 40.7127, lon = -74.0059), size = c(640, 640), zoom = 11, GRAYSCALE = TRUE)
#centered on Queens
gmap2 <- GetMap(center = c(lat = 40.7500, lon = -73.8667), size = c(640, 640), zoom = 11, GRAYSCALE = TRUE)
gmap3 <- GetMap(center = c(lat = 40.7500, lon = -73.8667), size = c(640, 640), zoom = 10, GRAYSCALE = TRUE)
```
### Make plots of overall coverage
 
```{r map_plots}
png("./NYC_subway_plots/all_stations_z11.png",width=640, height =640, res = 75)
PlotOnStaticMap(gmap2, lat = lats , lon = lons, cex=1.5,pch=20, col = "blue")
dev.off()

png("./NYC_subway_plots/all_stations_z10.png",width=640, height =640, res = 75)
PlotOnStaticMap(gmap3, lat = lats , lon = lons, cex=1.5,pch=20, col = "blue")
dev.off()

png("./NYC_subway_plots/all_staph_z11.png",width=640, height =640, res = 75)
PlotOnStaticMap(gmap2, lat = lats , lon = lons, cex=1.5,pch=20, col = cols_staph)
dev.off()

png("./NYC_subway_plots/all_staph_z10.png",width=640, height =640, res = 75)
PlotOnStaticMap(gmap3, lat = lats , lon = lons, cex=1.5,pch=20, col = cols_staph)
dev.off()
```

### PLot each CC type

```{r subtype_map_plots}

for (i in colnames(staph_df)[1:33]){
    plot_CC_types(CC = i, mat = staph_df, map10 = gmap3, map11 = gmap2, plotdir = "./NYC_subway_plots/", SRA_file = staph_df_coords)

}

```
### Mantel test for spatial autocorrelation of Staph aureus genotypes


```{r}
##First get Bray curtis matrix of dissimilarities
# braymat <- vegdist(staph_mat)
# jacmat <- vegdist(staph_mat, method = "jaccard")
# ##now get geographical distance
# 
# points_staph <- inner_join(NYCdata_SRA, Staph_betas, by = "Run") %>% select(Logitude,Latitude) 
# points_staph <- points_staph[-32,]#row 32 is mislabeled
# geogdist_staph <- dist(points_staph)
# set.seed(45678)
# mantel.rtest(geogdist_staph,braymat,nrepet = 9999)
# 
# mantel.rtest(geogdist_staph,jacmat,nrepet = 9999)
# 

```

### calculate hamming distances as an alternative using e1071 function

```{r hamming_analysis}
binmat <- bintr(coords_df_subtype_mat,0.2)
#hammmingmat <- hamming.distance(binmat)
bin_df <- as.data.frame(cbind(staph_df_coords$Run,binmat))
colnames(bin_df)[1] <- "Run"
# mantel.rtest(geogdist_staph,hammmingmat,nrepet = 9999)
run_geog <- select(staph_df_coords,Run, Latitude, Logitude)
combs <- combinations(r=2,v=run_geog$Run, n = length(run_geog$Run))
geog_distance_vector <- sapply(1:nrow(combs), function(x) dist_between_stations(combs[x,],run_geog))
hamm_distance_vector <- sapply(1:nrow(combs), function(x) H_distance_between_stations(combs[x,],bin_df))
#some were taken from the same location - need to filter these out
zero_stations <- which(geog_distance_vector == 0)
geog_distance_vector <- geog_distance_vector[-(zero_stations)]
hamm_distance_vector <- hamm_distance_vector[-(zero_stations)]
hist(geog_distance_vector, breaks = 50)
hist(hamm_distance_vector)
boxplot(geog_distance_vector ~ hamm_distance_vector, xlab = "Hamming distance", ylab = "km")
#try regression
reg <- lm(geog_distance_vector ~ as.numeric(hamm_distance_vector))
segments(x0=1,x1=7,y0=reg$coefficients[1],y1=reg$coefficients[1]+(reg$coefficients[2]*6), col = "red")
summary(reg)
```

#functions for looking at geog distance of individual subtypes by permutation
```{r}

CC_geog_perm_test(SRA_file = staph_df_coords, CC = "CC_8", cutoff = 0.2, s = 345, reps= 1000)
CC_geog_perm_test(SRA_file = staph_df_coords, CC = "CC_30", cutoff = 0.2, s = 4564, reps= 1000)
CC_geog_perm_test(SRA_file = staph_df_coords, CC = "CC_5", cutoff = 0.2, s = 23, reps= 1000)
CC_geog_perm_test(SRA_file = staph_df_coords, CC = "CC_45", cutoff = 0.2, s = 455, reps= 1000)
CC_geog_perm_test(SRA_file = staph_df_coords, CC = "CC_15", cutoff = 0.2, s = 989, reps= 1000)
CC_geog_perm_test(SRA_file = staph_df_coords, CC = "CC_1", cutoff = 0.2, s = 467856765, reps= 1000)
```



### session info

```{r}
sessionInfo()
```


